<robot name="primitive_chargepal_with_fix_plug">
  
  
  
  
  <link name="world"><inertial><mass value="0.0" /><origin rpy="0 0 0" xyz="0 0 0" /><inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" /></inertial></link>
  <joint name="world_joint" type="fixed">
    <parent link="world" />
    <child link="mobile_base_link" />
    <origin rpy="0.0 0.0 0.0" xyz="-0.275 0.0 0.2" />
  </joint>
  <link name="mobile_base_link">
    <inertial>
      <mass value="100.0" />
      <origin rpy="0 0 0" xyz="0 0 0.4" />
      <inertia ixx="4.136665012" ixy="0.0" ixz="0.0" iyy="6.666664000000002" iyz="0.0" izz="8.136663412" />
    </inertial>
    <visual>
      <material name="grey">
        <color rgba="0.5 0.5 0.5 1" />
      </material>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.8 0.58 0.4" />
      </geometry>
    </visual>
    <collision>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.8 0.58 0.4" />
      </geometry>
    </collision>
  </link>
  <joint name="tower_joint" type="fixed">
    <parent link="mobile_base_link" />
    <child link="tower_link" />
    <origin rpy="0.0 0.0 0.0" xyz="0.275 0.0 0.45" />
  </joint>
  <link name="tower_link">
    <inertial>
      <mass value="50.0" />
      <origin rpy="0 0 0" xyz="0 0 0.5" />
      <inertia ixx="2.443332356" ixy="0.0" ixz="0.0" iyy="1.3020828125000001" iyz="0.0" izz="1.6620826684999999" />
    </inertial>
    <visual>
      <material name="grey">
        <color rgba="0.5 0.5 0.5 1" />
      </material>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.25 0.58 0.5" />
      </geometry>
    </visual>
    <collision>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.25 0.58 0.5" />
      </geometry>
    </collision>
  </link>
  <joint name="battery_joint" type="fixed">
    <parent link="mobile_base_link" />
    <child link="battery_link" />
    <origin rpy="0.0 0.0 0.0" xyz="-0.15 0.0 0.4" />
  </joint>
  <link name="battery_link">
    <inertial>
      <mass value="300.0" />
      <origin rpy="0 0 0" xyz="0 0 0.4" />
      <inertia ixx="19.80061707975" ixy="0.0" ixz="0.0" iyy="11.562495375000003" iyz="0.0" izz="23.36311565475" />
    </inertial>
    <visual>
      <material name="aluminium">
        <color rgba="0.8 0.8 0.8 1" />
      </material>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.55 0.795 0.4" />
      </geometry>
    </visual>
    <collision>
      <orign rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0" />
      <geometry>
        <box size="0.55 0.795 0.4" />
      </geometry>
    </collision>
  </link>
  
  <gazebo reference="mobile_base_link">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="tower_link">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="battery_link">
    <material>Gazebo/Grey</material>
  </gazebo>
  
  <joint name="platform_ur10e_joint" type="fixed">
    <parent link="tower_link" />
    <child link="base_link" />
    <origin rpy="0.0 0.0 1.570795" xyz="0.0 0.0 0.25" />
  </joint>
  
  
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  
  <link name="base_link"><inertial><mass value="0.0" /><origin rpy="0 0 0" xyz="0 0 0" /><inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" /></inertial></link>
  <link name="base_link_inertia">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/base.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/base.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125" />
    </inertial>
  </link>
  <link name="shoulder_link">
    <visual>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/shoulder.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/shoulder.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="7.778" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.03147431257693659" ixy="0.0" ixz="0.0" iyy="0.03147431257693659" iyz="0.0" izz="0.021875624999999996" />
    </inertial>
  </link>
  <link name="upper_arm_link">
    <visual>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0.1762" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/upperarm.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0.1762" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/upperarm.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="12.93" />
      <origin rpy="0 1.5707963267948966 0" xyz="-0.306 0.0 0.175" />
      <inertia ixx="0.42175380379841093" ixy="0.0" ixz="0.0" iyy="0.42175380379841093" iyz="0.0" izz="0.03636562499999999" />
    </inertial>
  </link>
  <link name="forearm_link">
    <visual>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0.0393" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/forearm.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0.0393" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/forearm.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="3.87" />
      <origin rpy="0 1.5707963267948966 0" xyz="-0.285775 0.0 0.0393" />
      <inertia ixx="0.11079302548902206" ixy="0.0" ixz="0.0" iyy="0.11079302548902206" iyz="0.0" izz="0.010884375" />
    </inertial>
  </link>
  <link name="wrist_1_link">
    <visual>
      
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 -0.135" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/wrist1.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 -0.135" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/wrist1.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.005108247956699999" ixy="0.0" ixz="0.0" iyy="0.005108247956699999" iyz="0.0" izz="0.005512499999999999" />
    </inertial>
  </link>
  <link name="wrist_2_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.12" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/wrist2.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.12" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/wrist2.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.005108247956699999" ixy="0.0" ixz="0.0" iyy="0.005108247956699999" iyz="0.0" izz="0.005512499999999999" />
    </inertial>
  </link>
  <link name="wrist_3_link">
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 -0.1168" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/visual/wrist3.dae" />
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0" />
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 -0.1168" />
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10e/collision/wrist3.stl" />
      </geometry>
    </collision>
    <inertial>
      <mass value="0.202" />
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.025" />
      <inertia ixx="0.00014434577559500002" ixy="0.0" ixz="0.0" iyy="0.00014434577559500002" iyz="0.0" izz="0.00020452500000000002" />
    </inertial>
  </link>
  
  <joint name="base_link-base_link_inertia" type="fixed">
    <parent link="base_link" />
    <child link="base_link_inertia" />
    
    <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
  </joint>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link_inertia" />
    <child link="shoulder_link" />
    <origin rpy="0 0 0" xyz="0 0 0.1807" />
    <axis xyz="0 0 1" />
    <limit effort="330.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link" />
    <child link="upper_arm_link" />
    <origin rpy="1.570796327 0 0" xyz="0 0 0" />
    <axis xyz="0 0 1" />
    <limit effort="330.0" lower="-3.141592653589793" upper="0.0" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link" />
    <child link="forearm_link" />
    <origin rpy="0 0 0" xyz="-0.6127 0 0" />
    <axis xyz="0 0 1" />
    <limit effort="150.0" lower="-3.141592653589793" upper="0.0" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link" />
    <child link="wrist_1_link" />
    <origin rpy="0 0 0" xyz="-0.57155 0 0.17415" />
    <axis xyz="0 0 1" />
    <limit effort="56.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link" />
    <child link="wrist_2_link" />
    <origin rpy="1.570796327 0 0" xyz="0 -0.11985 -2.458164590756244e-11" />
    <axis xyz="0 0 1" />
    <limit effort="56.0" lower="-3.141592653589793" upper="3.141592653589793" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link" />
    <child link="wrist_3_link" />
    <origin rpy="1.570796326589793 3.141592653589793 3.141592653589793" xyz="0 0.11655 -2.390480459346185e-11" />
    <axis xyz="0 0 1" />
    <limit effort="56.0" lower="-0.7853981633974483" upper="5.497787143782138" velocity="3.141592653589793" />
    <dynamics damping="0" friction="0" />
  </joint>
  
  <link name="base"><inertial><mass value="0.0" /><origin rpy="0 0 0" xyz="0 0 0" /><inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" /></inertial></link>
  <joint name="base_link-base_fixed_joint" type="fixed">
    
    <origin rpy="0 0 3.141592653589793" xyz="0 0 0" />
    <parent link="base_link" />
    <child link="base" />
  </joint>
  
  <link name="flange"><inertial><mass value="0.0" /><origin rpy="0 0 0" xyz="0 0 0" /><inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" /></inertial></link>
  <joint name="wrist_3-flange" type="fixed">
    <parent link="wrist_3_link" />
    <child link="flange" />
    <origin rpy="0 -1.5707963267948966 -1.5707963267948966" xyz="0 0 0" />
  </joint>
  
  <link name="tool0"><inertial><mass value="0.0" /><origin rpy="0 0 0" xyz="0 0 0" /><inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" /></inertial></link>
  <joint name="flange-tool0" type="fixed">
    
    <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0 0 0" />
    <parent link="flange" />
    <child link="tool0" />
  </joint>
  
  <gazebo reference="shoulder_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="upper_arm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="forearm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_1_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_3_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_2_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="ee_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  
  
  
  
  <joint name="fts_fix" type="fixed">
    
    <origin rpy="0 0 0" xyz="0 0 0" />
    <parent link="tool0" />
    <child link="fts_robotside" />
  </joint>
  <link name="fts_robotside">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.01875" />
      <mass value="0.3" />
      
      <inertia ixx="0.000262" ixy="0" ixz="1" iyy="0.000265" iyz="1" izz="0.000219" />
    </inertial>
    <visual>
      <origin rpy="0 0 1.5707963267948966" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://chargepal_description/meshes/robotiq_ft_300/visual/robotiq_ft_300.stl" />
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1" />
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 1.5707963267948966" xyz="0 0 0" />
      <geometry>
        <mesh filename="package://chargepal_description/meshes/robotiq_ft_300/collision/robotiq_ft_300.stl" />
      </geometry>
    </collision>
  </link>
  <gazebo reference="fts_robotside">
    <material>Gazebo/FlatBlack</material>
  </gazebo>
  
  <joint name="measurement_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.01625" />
    <parent link="fts_robotside" />
    <child link="robotiq_force_torque_frame_id" />
  </joint>
  <link name="robotiq_force_torque_frame_id">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  
  <joint name="toolside_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.0375" />
    <parent link="fts_robotside" />
    <child link="fts_toolside" />
  </joint>
  <link name="fts_toolside">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  
  
  
  
  
  
  
  
  
  <joint name="plug_border_joint" type="fixed">
    <parent link="fts_toolside" />
    <child link="plug_border" />
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.015" />
  </joint>
  <link name="plug_border">
    <inertial>
      <mass value="0.1" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="5.812497675e-05" ixy="0.0" ixz="0.0" iyy="5.812497675e-05" iyz="0.0" izz="0.00010125" />
    </inertial>
    <visual>
      <material name="pink">
        <color rgba="0.937 0.109 0.882 1" />
      </material>
      <geometry>
        <cylinder length="0.03" radius="0.045" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.03" radius="0.045" />
      </geometry>
    </collision>
  </link>
  <joint name="plug_joint" type="fixed">
    <parent link="plug_border" />
    <child link="plug" />
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.055" />
  </joint>
  <link name="plug">
    <inertial>
      <mass value="0.25" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.00020772387524375" ixy="0.0" ixz="0.0" iyy="0.00020772387524375" iyz="0.0" izz="0.00014878125000000001" />
    </inertial>
    <visual>
      <material name="pink">
        <color rgba="0.937 0.109 0.882 1" />
      </material>
      <geometry>
        <cylinder length="0.08" radius="0.0345" />
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.08" radius="0.0345" />
      </geometry>
    </collision>
  </link>
  <joint name="plug_ref_frame_joint" type="fixed">
    <parent link="plug_border" />
    <child link="plug_reference_frame" />
    <axis xyz="0 0 0" />
    <origin rpy="0 0 0" xyz="0 0 0.015" />
  </joint>
  <link name="plug_reference_frame">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  <gazebo reference="handle">
    <material>Gazebo/ZincYellow</material>
  </gazebo>
  <gazebo reference="plug">
    <material>Gazebo/ZincYellow</material>
  </gazebo>
  <joint name="virt_tool_frame_x_joint" type="fixed">
    <parent link="plug_reference_frame" />
    <child link="virt_tool_frame_x" />
    <axis xyz="0 0 0" />
    <origin rpy="0 0 0" xyz="1 0 0" />
  </joint>
  <link name="virt_tool_frame_x">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  <joint name="virt_tool_frame_y_joint" type="fixed">
    <parent link="plug_reference_frame" />
    <child link="virt_tool_frame_y" />
    <axis xyz="0 0 0" />
    <origin rpy="0 0 0" xyz="0 1 0" />
  </joint>
  <link name="virt_tool_frame_y">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  <joint name="virt_tool_frame_z_joint" type="fixed">
    <parent link="plug_reference_frame" />
    <child link="virt_tool_frame_z" />
    <axis xyz="0 0 0" />
    <origin rpy="0 0 0" xyz="0 0 1" />
  </joint>
  <link name="virt_tool_frame_z">
    <inertial>
      <mass value="0.0" />
      <origin rpy="0 0 0" xyz="0 0 0" />
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0" />
    </inertial>
  </link>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <controlPeriod>0.001</controlPeriod>
      <legacyModeNS>false</legacyModeNS>
    </plugin>
  </gazebo>
</robot>